core/src/main/scala/org/apache/spark/rdd/RDD.scala

```diff
@@ -1521,7 +1521,7 @@ abstract class RDD[T: ClassTag](
       if (mapRDDs.partitions.length == 0) {
         Array.empty
       } else {
-        mapRDDs.reduce { (queue1, queue2) =>
+        mapRDDs.treeReduce { (queue1, queue2) =>
```

Comment:
I'm not sure if I miss anything, but this looks makes more sense based on the reason in the PR description.

Reply:
See what the test result is.